rclcpp_components/src/component_manager.cpp

```diff
@@ -0,0 +1,197 @@
+  const std::shared_ptr<LoadNode::Request> request,
+  std::shared_ptr<LoadNode::Response> response)
+{
+  (void) request_header;
+
+  std::string content;
+  std::string base_path;
+  if (!ament_index_cpp::get_resource(
+      "rclcpp_components", request->package_name, content, &base_path))
+  {
+    RCLCPP_ERROR(get_logger(), "Could not find requested resource in ament index");
+    response->error_message = "Could not find requested resource in ament index";
+    response->success = false;
+    return;
+  }
+
+  std::vector<std::string> lines = split(content, '\n', true);
+  for (auto line : lines) {
+    std::vector<std::string> parts = split(line, ';');
+    if (parts.size() != 2) {
+      RCLCPP_ERROR(get_logger(), "Invalid resource entry");
+      response->error_message = "Invalid resource entry";
+      response->success = false;
+      return;
+    }
+    // match plugin name with the same rmw suffix as this executable
+    if (parts[0] != request->plugin_name) {
+      continue;
+    }
+
+    std::string class_name = parts[0];
+    std::string fq_class_name = "rclcpp_components::NodeFactoryTemplate<" + class_name + ">";
+
+    // load node plugin
+    std::string library_path = parts[1];
+    if (!fs::path(library_path).is_absolute()) {
+      library_path = base_path + "/" + library_path;
+    }
+    class_loader::ClassLoader * loader;
+
+    if (loaders_.find(library_path) != loaders_.end()) {
+      loader = loaders_[library_path];
+    } else {
+      RCLCPP_INFO(get_logger(), "Load library %s", library_path.c_str());
+      try {
+        loader = new class_loader::ClassLoader(library_path);
+      } catch (const std::exception & ex) {
+        RCLCPP_ERROR(get_logger(), "Failed to load library: %s", ex.what());
+        response->error_message = "Failed to load library";
+        response->success = false;
+        return;
+      } catch (...) {
+        RCLCPP_ERROR(get_logger(), "Failed to load library");
+        response->error_message = "Failed to load library";
+        response->success = false;
+        return;
+      }
+    }
+    auto classes = loader->getAvailableClasses<rclcpp_components::NodeFactory>();
+    for (auto clazz : classes) {
+      RCLCPP_INFO(get_logger(), "Found class %s", class_name.c_str());
+      if (clazz == class_name || clazz == fq_class_name) {
+        RCLCPP_INFO(get_logger(), "Instantiate class %s", class_name.c_str());
+        loaders_[library_path] = loader;
+
+        auto node_factory = loader->createInstance<rclcpp_components::NodeFactory>(clazz);
+
+        std::vector<rclcpp::Parameter> parameters;
+        for (auto & p: request->parameters)
+        {
+          parameters.push_back(rclcpp::Parameter::from_parameter_msg(p));
+        }
+
+        auto options = rclcpp::NodeOptions()
+          .initial_parameters(parameters)
+          .arguments(request->remap_rules);
+
+        auto node_id = unique_id++;
+
+        node_wrappers_[node_id] = node_factory->create_node_instance(
```

Comment:
Something funky is going w.r.t. node naming, I get:

```sh
[component_container-1] [WARN] [rcl.logging_rosout]: Publisher already registered for provided node name. If this is due to multiple nodes with the same name then all logs for that logger name will go out over the existing publisher. As soon as any node with that name is destructed
 it will unregister the publisher, preventing any further logs for that name from being published on the rosout topic.
```

when loading the talker node via launch.

Reply:
Yeap, I can confirm that even though I request for a given node name, the fully qualified node name that the component gets once loaded is the same as that of the container.

Reply:
Try again with the new modifications, I should have been handling the name and namespace with remapping.

Reply:
Works perfectly now!